sfdx-source/apex-common/main/classes/fflib_Criteria.cls

```diff
@@ -0,0 +1,654 @@
+	{
+		return inSet(field, new Objects(new List<Long>(values)));
+	}
+	public fflib_Criteria inSet(Schema.SObjectField field, Set<String> values)
+	{
+		return inSet(field, new Objects(new List<String>(values)));
+	}
+	public fflib_Criteria inSet(Schema.SObjectField field, Objects values)
+	{
+		evaluators.add(new FieldSetEvaluator(field, fflib_Operator.INx, values));
+		return this;
+	}
+
+	/**
+	 * checks if the given sets contains the fields values
+	 *
+	 * @param field The field to use its value
+	 * @param values The value to be compared to the fields value
+	 *
+	 * @return Instance of fflib_Criteria
+	 *
+	 * @example
+	 *  new fflib_Criteria()
+	 *          .notInSet(Account.Type, new Set<Object>{'Customer', 'Competitor', 'Partner'})
+	 *
+	 * Evaluates:
+	 *      Account.Type NOT IN ('Customer','Competitor','Partner')
+	 */
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Date> values)
+	{
+		return notInSet(field, new Objects(new List<Date>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<DateTime> values)
+	{
+		return notInSet(field, new Objects(new List<DateTime>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Decimal> values)
+	{
+		return notInSet(field, new Objects(new List<Decimal>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Double> values)
+	{
+		return notInSet(field, new Objects(new List<Double>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Id> values)
+	{
+		return notInSet(field, new Objects(new List<Id>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Integer> values)
+	{
+		return notInSet(field, new Objects(new List<Integer>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Long> values)
+	{
+		return notInSet(field, new Objects(new List<Long>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<String> values)
+	{
+		return notInSet(field, new Objects(new List<String>(values)));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Set<Object> values)
+	{
+		return notInSet(field, new Objects(values));
+	}
+	public fflib_Criteria notInSet(Schema.SObjectField field, Objects values)
+	{
+		evaluators.add(new FieldSetEvaluator(field, fflib_Operator.NOT_IN, values));
+		return this;
+	}
+
+	/**
+	 * Generates the SOQL equivalent of the criteria provided
+	 *
+	 * @return The "where" part in the SOQL statement
+	 */
+	public String toSOQL()
+	{
+		if (this.evaluators.isEmpty()) return '';
+
+		String result = '';
+		for (Evaluator evaluator : this.evaluators)
+		{
+			if (result != '')
+			{
+				result += ' ' + this.comparatorType + ' ';
+			}
+			result += evaluator.toSOQL();
+		}
+
+		if (this.embraced) result = '(' + result + ')';
+
+		return result;
+	}
+
+	private static String operatorToString(fflib_Operator operator)
+	{
```

Comment:
Can this be a method (or map)  in the new fflib_Operator class?

Reply:
The fflib_Operator class is an Enum. 
In the original version (fflib2.0) there is a fflib_OperatorUtils class, with just only this method, not sure if it is that reuable to justify its own public class.

Reply:
It is a minor point, but it feels like the Operator class should control its string representations. Alternatively, you could say that various client/consumers should manage the rendering.